### PR TITLE
Support for OpenAI (GPT) + HuggingFace (Llama and crew)

### PR DESCRIPTION
**Pull request by allen-adastra**
Basic support so we can run `rag_server.py` with both GPT on the backend and `NousResearch/Nous-Hermes-Llama2-13b`

You'll likely want to create a conda environment from `environment.yaml` to use this from now on. Someone should test it.

### CODE REVIEW

File: rag/configs/huggingface.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+model_name: "NousResearch/Nous-Hermes-Llama2-13b"
+cache_dir: "/nobackup1/allenw/Scratch/" # Directory where the LLM will be cached.
```
Comment (virajmehta) [inline rag/configs/huggingface.yaml:2]: this should probably be an env var in the .env file

File: rag/llm_interface.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+FILE_PATH = os.path.dirname(os.path.realpath(__file__))
+
+class LLMInterface:
+    def __init__(self) -> None:
+        pass
+
+    def query(self, system: str, user_message: str):
+        raise NotImplementedError()
+
+class HuggingFaceInterface(LLMInterface):
```
Comment (virajmehta) [inline rag/llm_interface.py:17]: why do we need `model_interface` and `llm_interface` both?

**Review by virajmehta — COMMENTED** [review]
Generally it looks good, I think we can remove model_interface tho.

File: rag/llm_interface.py (hunk 2 above)
Comment (allen-adastra) [reply rag/llm_interface.py:17]: Already nuke `model_interface` in main

File: rag/configs/huggingface.yaml (hunk 1 above)
Comment (allen-adastra) [reply rag/configs/huggingface.yaml:2]: done

**Pull request merged**